modules/vector-sets/tests/vrange.py

```diff
@@ -0,0 +1,115 @@
+from test import TestCase, generate_random_vector
+import struct
+
+class BasicVRANGE(TestCase):
+    def getname(self):
+        return "VRANGE basic functionality and iteration"
+
+    def test(self):
+        # Add multiple elements with different names for lexicographical ordering
+        elements = [
+            "apple", "apricot", "banana", "cherry", "date",
+            "elderberry", "fig", "grape", "honeydew", "kiwi",
+            "lemon", "mango", "nectarine", "orange", "papaya",
+            "quince", "raspberry", "strawberry", "tangerine", "watermelon"
+        ]
+
+        # Add all elements to the vector set
+        for elem in elements:
+            vec = generate_random_vector(4)
+            vec_bytes = struct.pack('4f', *vec)
+            self.redis.execute_command('VADD', self.test_key, 'FP32', vec_bytes, elem)
+
+        # Test 1: Basic range with inclusive boundaries
+        result = self.redis.execute_command('VRANGE', self.test_key, '[apple', '[grape', '5')
+        result = [r.decode() for r in result]
+        assert result == ['apple', 'apricot', 'banana', 'cherry', 'date'], f"Expected first 5 elements from apple, got {result}"
+
+        # Test 2: Exclusive start boundary
+        result = self.redis.execute_command('VRANGE', self.test_key, '(apple', '[cherry', '10')
+        result = [r.decode() for r in result]
+        assert 'apple' not in result, "Exclusive boundary should not include 'apple'"
+        assert 'apricot' in result and 'banana' in result and 'cherry' in result, "Should include elements after apple up to cherry"
+
+        # Test 3: Exclusive end boundary
+        result = self.redis.execute_command('VRANGE', self.test_key, '[banana', '(cherry', '10')
+        result = [r.decode() for r in result]
+        assert 'banana' in result, "Should include banana"
+        assert 'cherry' not in result, "Exclusive end should not include cherry"
```

Comment:
Consider asserting the full reply.
The same comment applies to the cases listed below.